gcp_variant_transforms/vcf_to_bq.py

```diff
@@ -218,38 +219,64 @@ def run(argv=None):
       known_args.minimal_vep_alt_matching,
       counter_factory)
 
-  partitioner = variant_partition.VariantPartition()
+  partitioner = None
+  if known_args.optimize_for_large_inputs or known_args.partition_config_path:
+    partitioner = variant_partition.VariantPartition(
+        known_args.partition_config_path)
+
   beam_pipeline_options = pipeline_options.PipelineOptions(pipeline_args)
   pipeline = beam.Pipeline(options=beam_pipeline_options)
   variants = _read_variants(pipeline, known_args)
   variants |= 'FilterVariants' >> filter_variants.FilterVariants(
       reference_names=known_args.reference_names)
-  if variant_merger:
-    if known_args.optimize_for_large_inputs:
-      partitions = variants | 'PartitionVariants' >> beam.Partition(
-          partition_variants.PartitionVariants(partitioner),
-          partitioner.get_num_partitions())
-      merged = []
-      for i in xrange(partitioner.get_num_partitions()):
-        merged.append(partitions[i] | 'MergeVariants' + str(i) >>
+  if partitioner:
+    num_partitions = partitioner.get_num_partitions()
+    partitioned_variants = variants | 'PartitionVariants' >> beam.Partition(
+        partition_variants.PartitionVariants(partitioner), num_partitions)
+    variants = []
+    for i in range(num_partitions):
+      if partitioner.should_keep_partition(i):
+        variants.append(partitioned_variants[i])
+      else:
+        #  An extra dummy partition has been added for residual variants. This
```

Comment:
nit: I think this comment is too specific to the particular implementation logic inside partitioner (e.g. if we change the implementation of should_keep_partition, there shouldn't be any need to update the comment here). I think it should be something like `remove filtered partitions`, but I think it's obvious from the code itself so there is no need for the comment.